compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -87,7 +94,10 @@ public void process(SetMultimap<Class<? extends Annotation>, Element> elementsBy
   private void generateObjectGraph(TypeElement element) {
     DeclaredType component = MoreTypes.asDeclared(element.asType());
     ArrayList<ComponentMethodDescriptor> provisionMethods = new ArrayList<>();
-    ArrayList<ComponentMethodDescriptor> membersInjectionMethods = new ArrayList<>();
+    DirectedGraph<ComponentMethodDescriptor, DefaultEdge> memberInjectorsTypeGraph =
+        new DefaultDirectedGraph<>(DefaultEdge.class);
+    Map<TypeMirror, ComponentMethodDescriptor> memberInjectionMethodsTypeDescriptorMap = new HashMap<>();
```

Comment:
`TypeMirror`s need to be compared using `Types.isSameType()`, so they'd need to be wrapped into objects implementing `equals` in terms of `isSameType` to safely be able to use them as keys in a `Map`.